### PR TITLE
Rel 2.0.0

### PR DESCRIPTION
**Pull request by ethergreg**
Run away CPU if LogParser COM unreachable
Added lastSentTimeUTC
Diagnostics easier to parse

### CODE REVIEW

**Comment by chrisbaldauf** [comment]
I know we had talked about moving the version to 2.0, but I don't see any backward incompatible changes here - if we were to do a major version bump, I'd want to plan out what changes we'd include that could break backward compat. I also don't quite yet understand how the versioning of the chocolatey.org package works, because it clearly isn't pulling from the AssemblyInfo.

File: TimberWinR/Diagnostics/Diagnostics.cs [hunk 1]
```diff
@@ -52,29 +52,48 @@ private Assembly GetAssemblyByName(string name)
...
+                        new JProperty("logginglevel", LogManager.GlobalThreshold.ToString())
+                        )));
+            AddDiagnosis(json);
             return json;
         }
 
+        protected void AddDiagnosis(JObject wrapper)
+        {
+            wrapper.Add("inputs", GetDiagnosisByType("inputs", Manager.Listeners.ToList<IDiagnosable>()));
+            wrapper.Add("filters", GetDiagnosisByType("inputs", Manager.Config.Filters.ToList<IDiagnosable>()));
```
Comment (chrisbaldauf) [inline TimberWinR/Diagnostics/Diagnostics.cs]: copy / paste "inputs"?

File: TimberWinR/Inputs/IISW3CInputListener.cs [hunk 2]
```diff
@@ -145,6 +145,12 @@ private void IISW3CWatcher(string location)
                         catch (Exception ex)
                         {
                             LogManager.GetCurrentClassLogger().Error(ex);
+                            try
```
Comment (chrisbaldauf) [inline TimberWinR/Inputs/IISW3CInputListener.cs]: Why add the sleep in the catch here but in the finally in WindowsEvtInputListener? Having the wait in the finally seems safer...

File: TimberWinR/Outputs/Redis.cs [hunk 3]
```diff
@@ -132,6 +132,7 @@ public long SentMessages
         private long _errorCount;
         private long _redisDepth;
         private DateTime? _lastErrorTimeUTC;
+        private DateTime? _lastSentTimeUTC;
```
Comment (chrisbaldauf) [inline TimberWinR/Outputs/Redis.cs]: +1

**Comment by ethergreg** [comment]
merge this branch into rel-1.3.27.0

**Pull request closed without merging**